utils/validations/test_validate_sarif.go

```diff
@@ -0,0 +1,247 @@
+	for _, expectedRule := range expected.Tool.Driver.Rules {
+		rule, err := actual.GetRuleById(expectedRule.ID)
+		if !(assert.NoError(t, err, fmt.Sprintf("Run tool %s: Expected rule with ID %s not found", expected.Tool.Driver.Name, expectedRule.ID)) ||
+			assert.NotNil(t, rule, fmt.Sprintf("Run tool %s: Expected rule with ID %s not found", expected.Tool.Driver.Name, expectedRule.ID))) {
+			continue
+		}
+		validateSarifRule(t, exactMatch, expected.Tool.Driver.Name, expectedRule, rule)
+	}
+	// validate results
+	for _, expectedResult := range expected.Results {
+		result := getResultByResultId(expectedResult, actual.Results)
+		if !assert.NotNil(t, result, fmt.Sprintf("Run tool %s: Expected result with rule ID %s not found", expected.Tool.Driver.Name, sarifutils.GetResultRuleId(expectedResult))) {
+			continue
+		}
+		validateSarifResult(t, exactMatch, expected.Tool.Driver.Name, expectedResult, result)
+	}
+}
+
+func validateSarifRule(t *testing.T, exactMatch bool, toolName string, expected, actual *sarif.ReportingDescriptor) {
+	ValidateContent(t, exactMatch,
+		StringValidation{Expected: sarifutils.GetRuleFullDescription(expected), Actual: sarifutils.GetRuleFullDescription(actual), Msg: fmt.Sprintf("Run tool %s: Rule full description mismatch for rule %s", toolName, expected.ID)},
+		StringValidation{Expected: sarifutils.GetRuleFullDescriptionMarkdown(expected), Actual: sarifutils.GetRuleFullDescriptionMarkdown(actual), Msg: fmt.Sprintf("Run tool %s: Rule full description markdown mismatch for rule %s", toolName, expected.ID)},
+		StringValidation{Expected: sarifutils.GetRuleShortDescription(expected), Actual: sarifutils.GetRuleShortDescription(actual), Msg: fmt.Sprintf("Run tool %s: Rule short description mismatch for rule %s", toolName, expected.ID)},
+		StringValidation{Expected: sarifutils.GetRuleHelp(expected), Actual: sarifutils.GetRuleHelp(actual), Msg: fmt.Sprintf("Run tool %s: Rule help mismatch for rule %s", toolName, expected.ID)},
+		StringValidation{Expected: sarifutils.GetRuleHelpMarkdown(expected), Actual: sarifutils.GetRuleHelpMarkdown(actual), Msg: fmt.Sprintf("Run tool %s: Rule help markdown mismatch for rule %s", toolName, expected.ID)},
+	)
+	// validate properties
+	validateSarifProperties(t, exactMatch, expected.Properties, actual.Properties, toolName, expected.ID)
+}
+
+func getResultByResultId(expected *sarif.Result, actual []*sarif.Result) *sarif.Result {
+	for _, result := range actual {
+		if isPotentialSimilarResults(expected, result) && hasSameLocations(expected, result) {
+			return result
+		}
+	}
+	return nil
+}
+
+func isPotentialSimilarResults(expected, actual *sarif.Result) bool {
+	return sarifutils.GetResultRuleId(actual) == sarifutils.GetResultRuleId(expected) && sarifutils.GetResultMsgText(actual) == sarifutils.GetResultMsgText(expected) && sarifutils.GetResultProperty(sarifparser.WatchSarifPropertyKey, actual) == sarifutils.GetResultProperty(sarifparser.WatchSarifPropertyKey, expected)
```

Comment:
Should we compare by message? A text message is more likely to change in compare to ID or a property that we obtain
if you think it is not likely to break often we can leave it like that (or if it is absolutely necessary)

Reply:
in unit tests -> will not break
in integration -> if changes occur we will want to update our test case, it should not occur a lot and it is necessary so we will know if they stop returning this value (empty)